lms/djangoapps/student_account/views.py

```diff
@@ -72,6 +72,9 @@ def login_and_registration_form(request, initial_mode="login"):
     if request.user.is_authenticated():
         return redirect(redirect_to)
 
+    if 'ec_src' not in request.session and request.GET.get('ec_src', False):
```

Comment:
This seems like it could benefit from the revision mentioned above -- ie, moving the `ec_src` accessor into the `enterprise_helpers` module.